extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentPusherConfig.java

```diff
@@ -25,6 +25,8 @@
  */
 public class S3DataSegmentPusherConfig
 {
+  public static final int MAX_LISTING_LENGTH = 1000;
```

Comment:
`MAX_LISTING_LENGTH` is accessed in S3DataSegmentFinder.java#L71 to give the default value in case of `config.getMaxListingLength()` returns a value less than 0. Do you find this logic is not necessary?